### PR TITLE
[wip] add support for `bool` params

### PR DESCRIPTION
**Pull request by FaisalAbid**
I added support for bool params as well as bool query params.

Work in progress, would like your feedback & need to test if this works with discovery docs.

One thing I have not tested yet is if the document discovery works fine with this. What would be the best way to test this? 

I will also come up with a better example in toyApi.dart lol, I think the "hero" example I have is kinda dumb.

ref: https://github.com/dart-lang/rpc/issues/43

### CODE REVIEW

**Comment by wibling** [comment]
Hi Faisal,

Did you by any chance have time to finish this one off?

Best,
Gustav

**Comment by FaisalAbid** [comment]
Hey, sorry got caught up with a huge deadline, free as of tonight, so I'll wrap this up by tomorrow

**Comment by wibling** [comment]
Great! Thanks again for taking this on.

**Comment by FaisalAbid** [comment]
No problem! funny thing is the project i'm in crunch mode in is an api that uses rpc, have a few ideas to propose later on how to make rpc better

**Comment by wibling** [comment]
Glad to hear that! Looking forward to the ideas :)

**Comment by FaisalAbid** [comment]
Cool, so I've fixed the issues above, ran the generators and it seems like the generators generated valid code with the booleans. 

Check it out here

https://gist.github.com/FaisalAbid/7d47ad370da22f62392f

Next step is tests which you mentioned were a bit convoluted, so perhaps we merge this PR if its good, then I can investigate the tests as part of another PR

File: lib/src/config/method.dart [hunk 1]
```diff
@@ -97,6 +97,8 @@ class ApiConfigMethod {
                                   'path parameter: ${param.name}. '
                                   '${error.toString()}'), stack: stack);
         }
+      } else if(param.isBool){
```
Comment (wibling) [inline lib/src/config/method.dart]: NIT: space between if and (.

**Comment by wibling** [comment]
LGTM. Thanks again for fixing this. Merging it in.

File: lib/src/config/method.dart (hunk 1 above)
Comment (FaisalAbid) [inline lib/src/config/method.dart]: Ah don't know why my webstorm formatter didn't fix that. 

I'll make that fix with a future pr

Comment (wibling) [inline lib/src/config/method.dart]: No hurry:)

LMK if you run into issues with the tests. Also thanks for helping out with
the package. I am quite busy with other tasks at the moment and not as
responsive as I want to be, so your help is much appreciated :)

/gustav

Comment (FaisalAbid) [inline lib/src/config/method.dart]: No problem! Happy to help out. I'll get to tests either tonight or tomorrow.

Comment (wibling) [inline lib/src/config/method.dart]: Sounds good!

**Pull request merged**